Yank/yank.py

```diff
@@ -359,33 +355,66 @@ def _create_phase(self, thermodynamic_state, alchemical_phase):
+                    force.setExceptionParameters(exception, particle1, particle2, 0, sigma, epsilon)
+            else:
+                # Queue force to remove if not a NB fore
+                forces_to_remove.append(forceIndex)
+        # Remove all but NonbondedForce
+        # If done in preveious loop, nuber of forces change so indices change
+        for forceIndex in forces_to_remove[::-1]:
+            reference_system_LJ.removeForce(forceIndex)
+
+        reference_system_LJ_expanded = copy.deepcopy(reference_system_LJ)
         if is_periodic:
             # Determine minimum box side dimension
-            box_vectors = fully_interacting_system.getDefaultPeriodicBoxVectors()
+            box_vectors = reference_system_LJ_expanded.getDefaultPeriodicBoxVectors()
             min_box_dimension = min([max(vector) for vector in box_vectors])
 
             # Expand cutoff to minimize artifact and verify that box is big enough.
             # If we use a barostat we leave more room for volume fluctuations or
             # we risk fatal errors. If we don't use a barostat, OpenMM will raise
             # the appropriate exception on context creation.
-            # TODO: Make max_allowed_cutoff an option
             max_allowed_cutoff = 16 * unit.angstroms
+            # TODO: Make max_allowed_cutoff an option
             if thermodynamic_state.pressure and min_box_dimension < 2.25 * max_allowed_cutoff:
                 raise RuntimeError('Barostated box sides must be at least 36 Angstroms '
                                    'to correct for missing dispersion interactions')
 
             logger.debug('Setting cutoff for fully interacting system to maximum allowed (%s)' % str(max_allowed_cutoff))
-            for force_index in range(fully_interacting_system.getNumForces()):
-                force = fully_interacting_system.getForce(force_index)
-                if hasattr(force, 'setCutoffDistance'):
-                    force.setCutoffDistance(max_allowed_cutoff)
-                if hasattr(force, 'setCutoff'):
-                    force.setCutoff(max_allowed_cutoff)
 
-        # Construct thermodynamic state
-        fully_interacting_state = copy.deepcopy(thermodynamic_state)
-        fully_interacting_state.system = fully_interacting_system
+            # Expanded cutoff LJ system if needed
+            # We don't want to reduce the cutoff if its already large
+            for force in reference_system_LJ_expanded.getForces():
+                try:
+                    if force.getCutoffDistance() < max_allowed_cutoff:
+                        force.setCutoffDistance(max_allowed_cutoff)
+                except:
+                    pass
+                try:
+                    if force.getCutoff() < max_allowed_cutoff:
+                        force.setCutoff(max_allowed_cutoff)
+                except:
+                    pass
```

Comment:
Are these try/except clauses due to different OpenMM versions?

This is implementation works perfectly, but now that we only have `NonbondedForce`s at this point, we could even do something like
```python
for force in reference_system_LJ)expanded.getForces():
  try:
    if force.getCutoffDistance() < max_allowed_cutoff:
      force.setCutoffDistance(max_allowed_cutoff)
  except AttributeError:
    if force.getCutoff() < max_allowed_cutoff:
      force.setCutoff(max_allowed_cutoff)
```
This way if we messed something above this code we can catch it.